common/prettify.pm

```diff
@@ -353,6 +353,190 @@ sub Normal ($)
 ###############################################################################
 ###############################################################################
 
+package Prettify::Failed_Tests_HTML;
+
+use strict;
+use warnings;
+
+use FileHandle;
+
+###############################################################################
+
+sub new ($)
+{
+    my $proto = shift;
+    my $class = ref ($proto) || $proto;
+    my $self = {};
+    my $basename = shift;
+    my $buildname = shift;
+
+    my $path = substr($basename, 0, index($basename, '/'));
+    my $filename = $path . "/Failed_Tests.html";
+
+    $basename =~ s/^.*\///;
+
+    $self->{FULLHTML} = $basename . "_Full.html";
+    $self->{ERROR_COUNTER} = 0;
+    $self->{WARNING_COUNTER} = 0;
+    $self->{SECTION_COUNTER} = 0;
+    $self->{SUBSECTION_COUNTER} = 0;
+    $self->{TITLE} = "Failed Test Brief Log";
+    $self->{GIT_CHECKEDOUT_OPENDDS} = "unknown";
+
+    unless (-e $filename) {
+        my $file_handle = new FileHandle ($filename, 'w');
+        print {$file_handle} "<h1>$self->{TITLE}</h1>\n";
+    }
+
+    $self->{FH} = new FileHandle ($filename, '>>');
+    $self->{FILENAME} = $filename;
+    $self->{BUILDNAME} = $buildname;
+
+    bless ($self, $class);
+    return $self;
+}
+
+sub Header ()
+{
+    my $self = shift;
+
+    if (defined $self->{LAST_SECTION} && $self->{LAST_SECTION} eq 'Test') {
+        print {$self->{FH}} "<html>\n";
+        print {$self->{FH}} "<body bgcolor=\"white\">\n";
+    }
+}
+
+sub Footer ()
+{
+    my $self = shift;
+
+    if (defined $self->{LAST_SECTION} && $self->{LAST_SECTION} eq 'Test') {
+        # In the case where there was no errors or warnings, output a note
+        if ($self->{ERROR_COUNTER} == 0 && $self->{WARNING_COUNTER} == 0) {
+            print {$self->{FH}} "No Errors or Warnings detected<br>\n";
+        }
+
+        print {$self->{FH}} "</body>\n";
+        print {$self->{FH}} "</html>\n";
+    }
+}
+
+sub Section ($)
+{
+    my $self = shift;
+    my $s = shift;
+
+    # Escape any '<' or '>' signs
+    $s =~ s/</&lt;/g;
+    $s =~ s/>/&gt;/g;
+
+    my $counter = ++$self->{SECTION_COUNTER};
+
+    # Save for later use
+
+    $self->{LAST_SECTION} = $s;
+}
+
+sub Description ($)
+{
+    my $self = shift;
+
+    # Ignore
+}
+
+sub Timestamp ($)
+{
+    my $self = shift;
+    # Ignore
+}
+
+sub Subsection ($)
+{
+    my $self = shift;
+    my $s = shift;
+
+    # Escape any '<' or '>' signs
+    $s =~ s/</&lt;/g;
+    $s =~ s/>/&gt;/g;
+
+    my $counter = ++$self->{SUBSECTION_COUNTER};
+
+    # Save for later use
+
+    $self->{LAST_SUBSECTION} = $s;
+}
+
+sub Print_Sections ()
+{
+    my $self = shift;
+
+    if (defined $self->{LAST_SECTION} && defined $self->{LAST_SUBSECTION} && $self->{LAST_SECTION} eq 'Test') {
+        if (defined $self->{BUILDNAME}) {
+            print {$self->{FH}} "<hr><h2>$self->{BUILDNAME}</h2>\n";
+            my $rev = substr($self->{GIT_CHECKEDOUT_OPENDDS}, 0, 8);
```

Comment:
This should be more flexible, some scoreboards don't have OpenDDS at all, at that moment only printing the OpenDDS git revision looks to be confusing. As quick fix maybe only print it when the revision is not unknown, best would be to print all git commit revisions that are known at that point.